core/src/main/java/org/apache/carbondata/core/index/TableIndex.java

```diff
@@ -595,6 +595,30 @@ public void onEvent(Event event, OperationContext opContext) {
       throws IOException {
     List<Segment> segments = getCarbonSegments(allsegments);
     Map<String, Long> blockletToRowCountMap = new HashMap<>();
+    for (Segment segment : segments) {
+      List<CoarseGrainIndex> indexes = defaultIndex.getIndexFactory().getIndexes(segment);
+      for (CoarseGrainIndex index : indexes) {
+        if (null != partitions) {
+          // if it has partitioned index but there is no partitioned information stored, it means
+          // partitions are dropped so return empty list.
+          if (index.validatePartitionInfo(partitions)) {
+            return new HashMap<>();
+          }
+        }
+        index.getRowCountForEachBlock(segment, partitions, blockletToRowCountMap);
+      }
+    }
+    return blockletToRowCountMap;
+  }
+
+  /**
+   * Get the mapping of blocklet path and row count for all blocks. This method skips the
```

Comment:
can you please mention the count job of index server in the comment, as it's called only from index serve.

Reply:
done